doc/compiler/control/CheckpointRestoreCoordination.md

```diff
@@ -0,0 +1,434 @@
+
+First this method calls `J9::OptionsPostRestore::processOptionsPostRestore` to
+process any options set post-restore. Once this task is done, it acquires the
+Comp Monitor. It then resets the Checkpoint Status. Next it resets the Start
+time, and proceeds to invoke `resumeJITThreadsForRestore`, which
+1. Resumes the IProfiler Thread
+2. Resumes the Sampler Thread
+3. Resumes the Compilation Threads
+
+It then ensures that Swap Memory is availble before returning back to the VM.
+
+## Resetting the Start Time
+While the Checkpoint phase is conceptually part of building the application, in
+order to ensure consistency with parts of the compiler that memoize elapsd time,
+the start time is reset to pretend like the JVM started
+`persistentInfo->getElapsedTime()` milliseconds prior to time at restore.
+
+## Resuming the IProfiler Thread
+The Checkpointing Thread at this point has the Comp Monitor in hand.
+1. The Checkpointing Thread acquires the IProfiler Monitor
+2. The Checkpointing Thread sets the IProfiler Thread Lifetime State to
+   `TR_IProfiler::IPROF_THR_RESUMING`
+3. The Checkpointing Thread calls `notifyAll`, and releases the IProfiler
+   Monitor
+4. The IProfiler Thread resumes with the IProfiler Monitor in hand; it releases
+   the IProfiler Monitor, acquires the Comp Monitor, and then reacquires the
+   IProfiler Monitor. This is to maintain the
+   [coherency described above](#normal-execution-1).
+5. The IProfiler Thread sets the IProfiler Thread Lifetime State to
+   `TR_IProfiler::IPROF_THR_INITIALIZED`
+6. The Iprofiler releases the IProfiler Monitor and the Comp Monitor, and
+   returns to execute `processWorkingQueue`.
+
+## Resuming the Sampler Thread
+The Checkpointing Thread at this point has the Comp Monitor in hand.
+1. The Checkpointing Thread acquires the Sampler Monitor
+2. The Checkpointing Thread sets the Sampler Thread Lifetime State to
+   `TR_IProfiler::IPROF_THR_RESUMING`
+3. The Checkpointing Thread calls `notifyAll`, and releases the Sampler
+   Monitor
+4. The Sampler Thread resumes with the Sampler Monitor in hand; it releases
+   the Sampler Monitor, acquires the Comp Monitor, and then reacquires the
+   Sampler Monitor. This is to maintain the
+   [coherency described above](#normal-execution)
+5. The Sampler Thread sets the Sampler Thread Lifetime State to
+   `TR::CompilationInfo::SAMPLE_THR_INITIALIZED`
+5. The Sampler releases the Sampler Monitor and the Comp Monitor, and
+   returns to execute `samplerThreadProc`
+
+## Resuming Compilation Threads
+The Checkpointing Thread at this point has the Comp Monitor in hand. It simply
+invokes the existing `TR::CompilationInfo::resumeCompilationThread()` method,
+which ensures the appropriate amount of compilation threads are resumed (which
+could potentially be different from the number of threads pre-checkpoint because
+of the post-restore options).
+
+
+[^1]: As seen later on, even if a shutdown even occurs after this point, the
```

Comment:
`even if a shutdown even` sounds redundant. Do you mean `even if a shutdown event`?

Reply:
Yeah that was a typo